### PR TITLE
Create script to download provider binary

### PR DESCRIPTION
**Pull request by mu-bro**
Adding sh script for downloading terraform provider binary for current OS to `$HOME/.terraform.d/plugins` directory

### CODE REVIEW

File: download_binary.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+#
+# This script downloads terraform provider binary for current OS
+#
+############################################################################################
+
+RELEASE_VERSION=${1:-0.1.0}
+PROVIDER_NAME="terraform-provider-ec"
+TERRAFORM_PLUGIN_DIRECTORY="$HOME/.terraform.d/plugins"
+
+OS=$(uname -a | cut -d " " -f1 | awk '{print tolower($0)}')
```
Comment (omelnyk1) [inline download_binary.sh:13]: uname -s

File: download_binary.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+echo "Operation system: $OS"
+
+BINARY_NAME="${PROVIDER_NAME}_${OS}_amd64"
+BINARY_PATH="https://github.com/everonhq/terraform-provider-ec/releases/download/$RELEASE_VERSION/$BINARY_NAME"
+
+echo "Downloading binary plugin from $BINARY_PATH to $TERRAFORM_PLUGIN_DIRECTORY ..."
+
+mkdir $TERRAFORM_PLUGIN_DIRECTORY
```
Comment (RomanVlasenko) [inline download_binary.sh:21]: `mkdir -p` would be safer

File: download_binary.sh [hunk 3]
```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+############################################################################################
+#
+# This script downloads terraform provider binary for current OS
+#
+############################################################################################
+
+RELEASE_VERSION=${1:-0.1.0}
```
Comment (RomanVlasenko) [inline download_binary.sh:9]: forgot what means colon here. some kind of concatenation?

File: download_binary.sh (hunk 2 above)
Comment (omelnyk1) [inline download_binary.sh:21]: I'd recommend to use "mkdir -p ..." - no error if existing, make parent directories as needed

File: download_binary.sh (hunk 1 above)
Comment (RomanVlasenko) [inline download_binary.sh:13]: uname -s| awk '{print tolower($0)}' ?

File: download_binary.sh (hunk 3 above)
Comment (RomanVlasenko) [reply download_binary.sh:9]: should we update this line every time we increase the version?

File: download_binary.sh (hunk 1 above)
Comment (omelnyk1) [reply download_binary.sh:13]: I mean other argument in uname command)

File: download_binary.sh [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+OS=$(uname -a | cut -d " " -f1 | awk '{print tolower($0)}')
+echo "Operation system: $OS"
+
+BINARY_NAME="${PROVIDER_NAME}_${OS}_amd64"
+BINARY_PATH="https://github.com/everonhq/terraform-provider-ec/releases/download/$RELEASE_VERSION/$BINARY_NAME"
+
+echo "Downloading binary plugin from $BINARY_PATH to $TERRAFORM_PLUGIN_DIRECTORY ..."
+
+mkdir $TERRAFORM_PLUGIN_DIRECTORY
+wget -cO - $BINARY_PATH > $TERRAFORM_PLUGIN_DIRECTORY/$PROVIDER_NAME
```
Comment (RomanVlasenko) [inline download_binary.sh:22]: there's no wget on macos by default. would curl work?

Comment (omelnyk1) [inline download_binary.sh:22]: There is another approach with -O option. For instance:
`wget -O $TERRAFORM_PLUGIN_DIRECTORY/$PROVIDER_NAME $BINARY_PATH`

File: download_binary.sh [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+if [[ $? != 0 ]]; then
```
Comment (RomanVlasenko) [inline download_binary.sh:23]: won't it fail anyway if the exit code is not zero? perhaps this condition won't be reached unless you specify `set -e`?

File: download_binary.sh (hunk 3 above)
Comment (omelnyk1) [reply download_binary.sh:9]: `RELEASE_VERSION=${1:-0.1.0}` means set variable `RELEASE_VERSION` as a first argument in command line. If it wasn't specified - set 0.1.0

Comment (RomanVlasenko) [reply download_binary.sh:9]: then I'd extract $1 to a separate var, to make it more obvious

Comment (omelnyk1) [reply download_binary.sh:9]: Agree. Would be nice to have usage example

**Pull request merged**